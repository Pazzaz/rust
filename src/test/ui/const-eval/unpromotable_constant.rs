// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

#![allow(const_err)]

union Bad {
    usize: usize,
    ptr: &'static u32,
}

const FOO: usize = unsafe {
    Bad { ptr: &1 }.usize
};


fn main() {
    let x: &'static usize = &FOO; //~ ERROR does not live long enough
    let y: &'static usize = &(FOO % 42); //~ ERROR does not live long enough
}
