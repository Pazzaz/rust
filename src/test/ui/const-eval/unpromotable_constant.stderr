error[E0597]: borrowed value does not live long enough
  --> $DIR/unpromotable_constant.rs:24:30
   |
LL |     let x: &'static usize = &FOO; //~ ERROR does not live long enough
   |                              ^^^ temporary value does not live long enough
LL |     let y: &'static usize = &(FOO % 42); //~ ERROR does not live long enough
LL | }
   | - temporary value only lives until here
   |
   = note: borrowed value must be valid for the static lifetime...

error[E0597]: borrowed value does not live long enough
  --> $DIR/unpromotable_constant.rs:25:30
   |
LL |     let y: &'static usize = &(FOO % 42); //~ ERROR does not live long enough
   |                              ^^^^^^^^^^ temporary value does not live long enough
LL | }
   | - temporary value only lives until here
   |
   = note: borrowed value must be valid for the static lifetime...

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0597`.
